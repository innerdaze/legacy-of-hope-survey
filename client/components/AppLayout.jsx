import React, { Component } from 'react'
import Responsive from 'grommet/utils/Responsive'
import App from 'grommet/components/App'

class AppLayout extends Component {
  constructor(props) {
    super(props)

    this.state = {}

    this._onResponsive = this._onResponsive.bind(this)
  }

  componentDidMount() {
    this._responsive = Responsive.start(this._onResponsive)
  }

  componentWillUnmount() {
    this._responsive.stop()
  }

  _onResponsive(small) {
    this.setState({ small })
  }

  render() {
    return (
      <App centered={true}>
        { this.props.children }
      </App>
    )
  }
}

export default AppLayout
