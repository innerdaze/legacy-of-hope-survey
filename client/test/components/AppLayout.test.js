import React from 'react'
import AppLayout from '../../components/AppLayout'

let wrapper, children

describe('Testing on AppLayout...', () => {
  beforeEach(() => {
    children = [<div key='bar'>Foo</div>]
    wrapper = mount(<AppLayout children={children} />)
  })

  describe('AppLayout layout', () => {
    test('Expect AppLayout to have a parent App comp', () => {
      expect(wrapper.find('App')).toHaveLength(1)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('Test on methods', () => {
    test('Expect componentWillUnmount is called when AppLayout is going to destroy', () => {
      let spy = jest.spyOn(wrapper.instance(), 'componentWillUnmount')

      wrapper.unmount()
      expect(spy).toHaveBeenCalled()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
