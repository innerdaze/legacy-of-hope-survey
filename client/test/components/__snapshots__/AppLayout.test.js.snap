// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on AppLayout... AppLayout layout Expect AppLayout to have a parent App comp 1`] = `
<AppLayout>
  <App
    centered={true}
  >
    <div
      className="grommet grommetux-app grommetux-app--centered"
      lang="en-US"
    >
      <div
        key="bar"
      >
        Foo
      </div>
      <SkipLinks>
        <Layer
          align="top"
          hidden={true}
          id="skip-link-layer"
        >
          <span
            style={
              Object {
                "display": "none",
              }
            }
          />
        </Layer>
      </SkipLinks>
      <div
        aria-live="polite"
        className="grommetux-app__announcer"
      />
    </div>
  </App>
</AppLayout>
`;

exports[`Testing on AppLayout... Test on methods Expect componentWillUnmount is called when AppLayout is going to destroy 1`] = `""`;
